### PR TITLE
chore(deps): update to backstage 1.22

### PR DESCRIPTION
**Pull request by schultzp2020**
## Description

Updates the showcase to backstage 1.22.1.

## Which issue(s) does this PR fix

- Fixes https://issues.redhat.com/browse/RHIDP-902
- Fixes https://github.com/janus-idp/backstage-showcase/issues/874
- Fixes https://github.com/janus-idp/backstage-showcase/issues/886

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: packages/backend/src/index.ts [hunk 1]
```diff
@@ -44,13 +44,7 @@ import {
   gatherDynamicPluginsSchemas,
 } from './schemas';
 
-// TODO(davidfestal): The following import is a temporary workaround for a bug
-// in the upstream @backstage/backend-plugin-manager package.
-//
-// It should be removed as soon as the upstream package is fixed and released.
-// see https://github.com/janus-idp/backstage-showcase/pull/600
 import { WinstonLogger } from '@backstage/backend-app-api';
```
Comment (Zaperex) [inline packages/backend/src/index.ts:47]:
```suggestion
```
Remove this and move merge it with the imports from `@backstage/backend-app-api` at the top of the file. Can't seem to select it though in the review panel.

File: dynamic-plugins/wrappers/backstage-plugin-azure-devops-backend-dynamic/dist-dynamic/package.json [hunk 2]
```diff
@@ -44,10 +45,14 @@
   ],
   "bundleDependencies": true,
   "peerDependencies": {
-    "@backstage/backend-common": "0.19.8",
-    "@backstage/backend-plugin-api": "0.6.6",
-    "@backstage/backend-plugin-manager": "npm:@janus-idp/backend-plugin-manager@0.0.2-janus.5",
```
Comment (nickboldt) [inline dynamic-plugins/wrappers/backstage-plugin-azure-devops-backend-dynamic/dist-dynamic/package.json:49]: +1 for getting rid of these `npm:` dependencies in favour of properly declared ones

File: dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-github-dynamic/dist-dynamic/package.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
     "test": "backstage-cli package test --passWithNoTests --coverage",
     "clean": "backstage-cli package clean",
     "clean-dynamic-sources": "yarn clean && rm -Rf node_modules",
-    "export-dynamic": "janus-cli package export-dynamic-plugin --embed-package @backstage/plugin-catalog-backend-module-github"
+    "export-dynamic": "janus-cli package export-dynamic-plugin --embed-package @backstage/plugin-catalog-backend-module-github --override-interop default"
```
Comment (nickboldt) [inline dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-github-dynamic/dist-dynamic/package.json:29]: I take it this was missing from some of the dynamic plugins, but should be there on all of them?

File: dynamic-plugins/wrappers/backstage-plugin-dynatrace/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "backstage-plugin-dynatrace",
-  "version": "7.0.5",
+  "version": "8.0.3",
```
Comment (nickboldt) [inline dynamic-plugins/wrappers/backstage-plugin-dynatrace/package.json:3]: major version bump here: any concerns?

File: e2e-tests/package.json [hunk 5]
```diff
@@ -15,14 +15,14 @@
     "lint": "eslint . --ext .js,.ts"
   },
   "devDependencies": {
-    "@cypress/eslint-plugin-dev": "^5.3.2",
+    "@cypress/eslint-plugin-dev": "5.3.2",
```
Comment (nickboldt) [inline e2e-tests/package.json:18]: intentional pinning to specific single versions instead of `^` versions?

File: package.json [hunk 6]
```diff
@@ -54,8 +54,8 @@
     "undici": "5.26.5"
   },
   "resolutions": {
-    "@types/react": "17.0.69",
-    "@types/react-dom": "17.0.22"
+    "@types/react": "18",
```
Comment (nickboldt) [inline package.json:57]: react 17 -> 18... any concerns here?

File: dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-github-dynamic/dist-dynamic/package.json (hunk 3 above)
Comment (davidfestal) [reply dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-github-dynamic/dist-dynamic/package.json:29]: It's on the wrappers.
In fact this was already set in the main package. The fact that it is added in the `dist-dynamic/package.json` is just the consequence that this chane had not been committed back the last time.

File: e2e-tests/package.json (hunk 5 above)
Comment (davidfestal) [reply e2e-tests/package.json:18]: I think it's part of the way the version-bumping script works, isn't it @schultzp2020 ?

**Comment by davidfestal** [comment]
OK for me as well.
We should just check whether the Showcase tests on OCP failure is meaningful. cc @josephca

File: dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-github-dynamic/dist-dynamic/package.json (hunk 3 above)
Comment (schultzp2020) [reply dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-github-dynamic/dist-dynamic/package.json:29]: @davidfestal

File: dynamic-plugins/wrappers/backstage-plugin-dynatrace/package.json (hunk 4 above)
Comment (schultzp2020) [reply dynamic-plugins/wrappers/backstage-plugin-dynatrace/package.json:3]: I'll double check.

File: e2e-tests/package.json (hunk 5 above)
Comment (schultzp2020) [reply e2e-tests/package.json:18]: That's been our rule in the showcase repo for a while. It must have been added while I was out.

**Comment by invincibleJai** [comment]
I verified this locally the instance came up and could add entities and tried Tekton, Topology and quay plugin.

I observed on `yarn run export-dynamic` I see some version updates in package.json under the wrapper, is this expected or ?

image

**Comment by schultzp2020** [comment]
> Thanks @schultzp2020 , I verified this locally the instance came up and could add entities and verified Tekton, Topology, and quay plugins 🎉
> 
> I observed on `yarn run export-dynamic` I see some version updates in package.json under the wrapper, is this expected or ?
> 
> image

Yes, the version increase is expected.

File: e2e-tests/package.json (hunk 5 above)
Comment (josephca) [reply e2e-tests/package.json]: We recently [migrated from cypress to playwright](https://github.com/janus-idp/backstage-showcase/pull/851). And, this issue should be resolved when you [rebase](https://github.com/janus-idp/backstage-showcase/blob/main/e2e-tests/package.json).

**Comment by josephca** [comment]
> OK for me as well. We should just check whether the Showcase tests on OCP failure is meaningful. cc @josephca

You may ignore the PR failure for now. @gustavolira is currently migrating from cypress to playwright and he's checking the PR test failure closely.

**Pull request merged**